zerver/lib/zcommand.py

```diff
@@ -1,14 +1,40 @@
-from typing import Any, Dict
+from typing import Any, Dict, Tuple, Optional
 from django.utils.translation import ugettext as _
 
-from zerver.models import UserProfile
+from zerver.models import UserProfile, get_realm_stream, Stream
 from zerver.lib.actions import do_set_user_display_setting
 from zerver.lib.exceptions import JsonableError
+from zerver.lib.bugdown import MentionData
+
+import re
+
+SUBSCRIBE_REGEX = r'^subscribe[ ]+#\*\*([^*]+)\*\*[ ]+(@\*\*[^*]+\*\*)$'
+
+def get_user_and_stream_subscribe_command(content: str,
+                                          realm_id: int) -> Tuple[Optional[Stream],
+                                                                  Optional[int]]:
+    stream, user_id = None, None
+    match = re.search(SUBSCRIBE_REGEX, content)
+    if match:
+        stream_name, user_mention = match.group(1), match.group(2)
+
+        try:
+            stream = get_realm_stream(stream_name=stream_name, realm_id=realm_id)
+        except Stream.DoesNotExist:
+            pass
```

Comment:
I was unsure about extracting the function in the `/mute_topic` commit, but it can be re-used nicely here, depending which PR is merged first.

Reply:
Right, that's my plan.